pkg/client/keycloak/client.go

```diff
@@ -175,6 +175,17 @@ func (kc *kcClient) GetToken() (string, error) {
 	return tokenResp.AccessToken, nil
 }
 
+// IsJWTTokenExpired returns false if JWT token is not expired, otherwise returns false
+// This method does NOT validate the token, and should not be used when validation is necessary
+func IsJWTTokenExpired(accessToken string) bool {
+	if token, _ := jwt.Parse(accessToken, nil); token != nil {
+		tokenClaims := token.Claims.(jwt.MapClaims)
+		exp := time.Unix(int64(tokenClaims["exp"].(float64)), 0)
```

Comment:
expTime would be better a variable name. WDYT?

Reply:
better to separate this as 2  steps and validate that expiry time is present, before performing any operations on it. According to the RFC: https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.4 use of this claim is optional, so better to check if exp is there, and valid and get it and then convert/typecast it to time.

Reply:
Good point @Rajagopalan-Ranganathan, thanks
@akoserwal  what are your thoughts on this? Would passing an empty exp return true?

Reply:
exp would be present in every jwt token. if it is not present than return true

Reply:
Thanks for the clarification @akoserwal. How is this?

```golang
func IsJWTTokenExpired(accessToken string) bool {
	if token, _ := jwt.Parse(accessToken, nil); token != nil {
		tokenClaims := token.Claims.(jwt.MapClaims)
		if _, ok := tokenClaims["exp"]; ok {
			expTime := time.Unix(int64(tokenClaims["exp"].(float64)), 0)
			return time.Now().After(expTime)
		}
	}
	return true
}
```